src/new_index/schema.rs

```diff
@@ -838,7 +851,16 @@ impl ChainQuery {
     pub fn lookup_raw_txn(&self, txid: &Txid, blockhash: Option<&BlockHash>) -> Option<Bytes> {
         let _timer = self.start_timer("lookup_raw_txn");
 
-        if self.light_mode {
+        if let Ok(cache) = self.txs_cache.lock() {
```

Comment:
I don't think is expected to fail, but since here everything would work normally in case of failure (just perf degradation) I preferred this way

Reply:
A poisoned mutex is likely to indicate some underlying coding issue, wouldn't it be better to let it error visibly?

I would at least log a warn-level message instead of silently ignoring it, but generally my preference is the "fail fast" approach for unanticipated errors. Failing fast (and restarting) would also re-enable the tx cache, instead of having it continue with degraded performance (until the process eventually restarts for another reason).